bsp/Src/ADC.c

```diff
@@ -0,0 +1,163 @@
+#include "ADC.h"
+
+// Define Queue Handles
+#ifdef ADC1
+static ADC_HandleTypeDef hadc1_ = {.Instance = ADC1};
+ADC_HandleTypeDef* hadc1 = &hadc1_;
+#endif
+
+#ifdef ADC2
+static ADC_HandleTypeDef hadc2_ = {.Instance = ADC2};
+ADC_HandleTypeDef* hadc2 = &hadc2_;
+#endif
+
+#ifdef ADC3
+static ADC_HandleTypeDef hadc3_ = {.Instance = ADC3};
+ADC_HandleTypeDef* hadc3 = &hadc3_;
+#endif
+
+// Stores queue to push val to
+QueueHandle_t* adc1_q;
+QueueHandle_t* adc2_q;
+QueueHandle_t* adc3_q;
+
+adc_status_t ADC_Init(ADC_InitTypeDef init, uint8_t adcNum) {
+    // Power ADC clocks and set configs
+    switch (adcNum) {
+        case 1:
+            __HAL_RCC_ADC1_CLK_ENABLE();
+            hadc1->Init = init;
+            if (HAL_ADC_Init(hadc1) != HAL_OK) return ADC_INIT_FAIL;
+            break;
+        case 2:
+            __HAL_RCC_ADC2_CLK_ENABLE();
+            hadc2->Init = init;
+            if (HAL_ADC_Init(hadc2) != HAL_OK) return ADC_INIT_FAIL;
+            break;
+        case 3:
+            __HAL_RCC_ADC3_CLK_ENABLE();
+            hadc3->Init = init;
+            if (HAL_ADC_Init(hadc3) != HAL_OK) return ADC_INIT_FAIL;
+            break;
+        default:
+            // Couldn't initalize any ADC
+            return ADC_INIT_FAIL;
+            break;
+    }
+
+    // Configure NVIC
+    HAL_NVIC_SetPriority(ADC_IRQn, 5, 0);
+    HAL_NVIC_EnableIRQ(ADC_IRQn);
+    
+    return ADC_OK;
+}
+
+
+adc_status_t ADC_OneShotRead(uint32_t channel, uint32_t samplingTime, uint8_t adcNum, QueueHandle_t* q) {
+    if (xQueueIsQueueFullFromISR(*q)) {
```

Comment:
This isn't being called from an ISR so you should prolly just check uxQueueSpacesAvailable() == 0

Reply:
👍